src/py-opentimelineio/opentimelineio/console/otiotool.py

```diff
@@ -861,8 +870,29 @@ def summarize_timeline(list_tracks, list_clips, list_media, verify_media,
             if list_tracks:
                 print(f"TRACK: {child.name} ({child.kind})")
         if isinstance(child, otio.schema.Clip):
-            if list_clips:
-                print("  CLIP:", child.name)
+            if list_clips or verify_ranges:
+                if verify_ranges:
+                    range_msg = ""
+                    try:
+                        source = child.source_range
+                        available = child.available_range()
+
+                        src_start = source.start_time.value
+                        src_dur = source.duration.value
+                        avail_start = available.start_time.value
+                        avail_dur = available.duration.value
+
+                        starts_early = src_start < avail_start
+                        ends_late = (src_dur - src_start) > (avail_dur - avail_start)
```

Comment:
I think you can use the time algebra to answer this question... see:

https://github.com/AcademySoftwareFoundation/OpenTimelineIO/blob/42158ebd33eb4ddf5dc2843a63998b4d8523be54/src/opentime/timeRange.h#L145

I think what you want is available_range contains source_range

Reply:
Good call, the `contains()` didn't seem to work for me initially when the source range matched available range exactly. But that'd be an easy check

Reply:
So it turns out `contains()` uses `end_time_exclusive()`, which means it fails if the source and available ranges end at exactly the same time (such as using the entire available range). But using `end_time_inclusive()` drastically simplifies the time algebra. Pushed the change.

Reply:
@ssteinbach this is the same issue that @reinecke and I talked about a while back (on Slack I think?). There is/was a suggestion to add a convenience for this case where `contains()` doesn't behave as people expect. I think we can leave the code in this PR as-is and address that API change some other time.